packages/salesforcedx-vscode-apex/src/index.ts

```diff
@@ -52,9 +52,11 @@ export async function activate(context: vscode.ExtensionContext) {
     .then(async () => {
```

Comment:
@lcampos I've been meaning to ask this general question. It seems strange that we mix `promise.then()...` with async await. When we are making changes to a method/file where we see both being used, is it ok to refactor the promise chain to async await?

Reply:
It really depends on what you're trying to achieve. Using `await` will cause the execution to pause until the promise is resolved. In this case we chose to wait for the language server to be created (line 45) but we do not want to wait for it to be ready and be done parsing the project, doing that would mean the activation of the extension is delayed.

I'd say that if you need to have the value from the promise to continue executing then rest of the code (synchronous) then you should use await, if not and it's expensive/time consuming then having it being asynchronous is the way to go.

Reply:
I like that idea. As another data point, we should also see if we are _over_ awaiting and making things serial. If we are, then we should use https://developers.google.com/web/fundamentals/primers/async-functions#careful_avoid_going_too_sequential